File: app/src/main/java/com/love/awaker/imageloader/ImageLoader.java
package com.love.awaker.imageloader;

/**
 * Copyright ©2017 by Teambition
 */

public final class ImageLoader {

    private static GlideImpl GLIDE;

    private ImageLoader() {}

    public static IImageLoader get() {
        if (GLIDE == null) {
            synchronized (ImageLoader.class) {
                if (GLIDE == null) {
                    GLIDE = new GlideImpl();
                }
            }
        }
        return GLIDE;
    }
}

File: imageloader/src/main/java/com/future/imageloader/glide/ImageLoader.kt
package com.future.imageloader.glide

class ImageLoader {

    companion object {

        @Volatile
        private var INSTANCE: IImageLoader? = null

        @JvmStatic
        fun get(): IImageLoader = INSTANCE ?: synchronized(ImageLoader::class) {
            INSTANCE ?: GlideImpl().also { INSTANCE = it }
        }
    }
}